### PR TITLE
Display any existing customer account credit balance in the first positive invoice

### PR DESCRIPTION
**Pull request by paulbrown1982**
Tweaks to use the new billingSchedule API which now takes an account as it applies any existing credit balance to the first positive invoice.

Requires: https://github.com/guardian/membership-common/pull/410

cc @johnduffell @pvighi @jacobwinch @AWare @jayceb1 

picture 2

### CODE REVIEW

**Comment by pvighi** [comment]
LGTM 👍

File: app/controllers/AccountManagement.scala [hunk 1]
```diff
@@ -78,12 +78,12 @@ object ManageDelivery extends ContextLogging {
       form <- EitherT(Future.successful(SuspendForm.mappings.bindFromRequest().value \/> "Please check your selections and try again"))
       maybeDeliverySub <- EitherT(SessionSubscription.subscriptionFromRequest.map(_ \/> "Could not find an active subscription"))
       sub <- EitherT(Future.successful(maybeDeliverySub.asDelivery \/> "Is not a Home Delivery subscription"))
-      billCycleDay <- EitherT(tpBackend.zuoraService.getAccount(sub.accountId).map(_.billCycleDay).map(intToEither))
+      account <- EitherT(tpBackend.zuoraService.getAccount(sub.accountId).map(toScalaZRight))
```
Comment (johnduffell) [inline app/controllers/AccountManagement.scala]:
```scala
account <- EitherT(tpBackend.zuoraService.getAccount(sub.accountId).map(toScalaZRight))
```
could be
```scala
account <- EitherT(tpBackend.zuoraService.getAccount(sub.accountId).map(\/-.apply))
```
or am I missing something?

Comment (johnduffell) [inline app/controllers/AccountManagement.scala]: ah maybe the type of the left is also important?

File: app/views/support/BillingScheduleOps.scala [hunk 2]
```diff
@@ -5,6 +5,6 @@ import scalaz.syntax.std.boolean._
 object BillingScheduleOps {
   implicit class InvoiceOps(bill: Bill) {
     def isDiscounted(str: String): Option[String] =
-      bill.items.list.exists(_.amount < 0).option(str)
+      (bill.accountCredit.isDefined || bill.items.list.exists(_.amount < 0)).option(str)
```
Comment (johnduffell) [inline app/views/support/BillingScheduleOps.scala]: isDiscounted doesn't return a boolean, which is surprising

Comment (johnduffell) [inline app/views/support/BillingScheduleOps.scala]: I'd prefer to separate the boolean determining logic from the massaging types, so even if we want to keep the behaviour of this method, put it into a val and then use the val on the next line

**Comment by johnduffell** [comment]
looks good, just a small comment about naming. I think the first comment might be just be the compiler needing keeping happy. 👍

**Pull request merged**